src/Models/Apis.ts

```diff
@@ -39,3 +42,22 @@ export async function checkAzureLogin(): Promise<boolean> {
 
   return true;
 }
+
+export async function checkExtensionAvailable(extensionName: ExtensionName): Promise<boolean> {
+  if (!getExtension(extensionName)) {
+    const choice = await vscode.window.showInformationMessage(
+      `${extensionName} extension is required for the current project. Do you want to install it from marketplace?`,
+      "Yes",
+      "No"
+    );
+    if (choice === "Yes") {
+      vscode.commands.executeCommand(
+        VscodeCommands.VscodeOpen,
+        vscode.Uri.parse(("vscode:extension/" + ExtensionNameIdMap[ExtensionName.Arduino]) as string)

```

Comment:
I think it is extensionName instead of ExtensionName.Arduino here?

Reply:
Fixed.